salt/modules/file.py

```diff
@@ -3499,13 +3499,21 @@ def _get_local_file_source_sum(path):
     # If we have a source defined, let's figure out what the hash is
     if source:
         urlparsed_source = _urlparse(source)
-        if urlparsed_source.scheme == 'salt':
+        parsed_scheme = urlparsed_source.scheme
+        parsed_path = os.path.join(
+                urlparsed_source.netloc, urlparsed_source.path).rstrip(os.sep)
+
+        if parsed_scheme.lower() in 'abcdefghijklmnopqrstuvwxyz':
```

Comment:
Also, this can be done in a less error-prone way using [string.ascii_lowercase](https://docs.python.org/2/library/string.html?highlight=string.ascii_lowercase#string.ascii_lowercase).